### PR TITLE
Added member info page

### LINKED ISSUE #302: “QR redirija a una vista de datos del usuario”, (esto con el fin de verificar al ingreso del evento que sea asistente).
**Explicación**

Al asistir a eventos se desea verificar que los asistentes sean socios o personas registradas previamente en DID, se desea al escanear obtener los datos de el socio y saber si su membrecía esta activa, estos datos pueden ser solicitados por personas que no son socios de DID.

**- Resultado actual:**

El QR redirige al panel administrativo sin ningún dato especifico

**- Resultado esperado:**

Redirigir al una vista en el panel administrativo que muestre los datos del socio.

**Usar como referencia esta imagen**

Group 6

**Flujo esperado**

1. cuando el usuario este en la aplicación este ingresara a su perfil allí esta disponible el código QR
2. Al escanear Redirecciona al navegador (Se debe tener presente que el que escanea puede no ser usuario o socio de DID, este tendrá acceso a los datos del socio por el QR).
3. Podrá ver la imagen de socio y sus datos.

### PR DESCRIPTION
**Pull request by e-azocar**
### Description

I added member info page
image

fixes #302 

#### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How to test it

If applicable, describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

Go to '/info/{Member code}'. For instance '/info/ZZ00000003'. Or scan member user QR code from app

### Checklist:

- [x] Doesn't break the current code.
- [x] Passes linters and test, also is building.
- [x] Doesn't have spelling or grammatical problems.
- [x] Doesn't have unnecessary comments or debugging code.
- [x] The branch is updated with the last dev branch changes.

### CODE REVIEW

**Review by Eghinner — APPROVED** [review]

File: .eslintrc.js [hunk 1]
```diff
@@ -82,6 +82,7 @@ module.exports = {
       },
     ],
     'import/core-modules': 0,
+    '@next/next/no-img-element': 0,
```
Comment (Israel-Laguan) [inline .eslintrc.js]: Why not use false? Using 0 is confusing

File: .stylelintrc.json [hunk 2]
```diff
@@ -24,7 +24,8 @@
         ],
         "ignoreSelectors": [":export"]
       }
-    ]
+    ],
+    "selector-max-compound-selectors": null
```
Comment (Israel-Laguan) [inline .stylelintrc.json:28]: What about using false

File: controllers/members.ts [hunk 3]
```diff
@@ -19,6 +19,21 @@ interface GetParams {
...
+export const getMemberByMemberCode = async (memberCode: string) => {
+  const { data } = await client.query<{ users: User[] }>({
+    query: USER_BY_MEMBER_CODE,
+    variables: {
+      memberCode,
+    },
+  })
+
+  if (data.users.length) {
+    return { user: data.users[0], error: null }
```
Comment (Israel-Laguan) [inline controllers/members.ts]:
```suggestion
    return { user: data.users[0] }
```

File: controllers/members.ts [hunk 4]
```diff
@@ -19,6 +19,21 @@ interface GetParams {
...
+  }
+
+  return { error: 'Usuario no encontrado', user: null }
```
Comment (Israel-Laguan) [inline controllers/members.ts]:
```suggestion
  return { error: 'Usuario no encontrado' }
```

I think the other way is more semantic, first check if there is an error, is called "early return"

File: pages/info/[memberCode].tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+import styles from '../../styles/MemberInfo.module.scss'
+
+interface MemberInfoProps {
+  user: User
+  error: String
+}
+
+const MemberInfo: NextPage<MemberInfoProps> = ({ user, error }) => {
+  return (
```
Comment (Israel-Laguan) [inline pages/info/[memberCode].tsx:17]:
```suggestion
const MemberInfo: NextPage<MemberInfoProps> = ({ user, error }) => (
```

File: pages/info/[memberCode].tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        </p>
+      </div>
+    </div>
+  )
+}
+
+export default MemberInfo
+
+export const getServerSideProps: GetServerSideProps = async context => {
+  const { memberCode } = context.query
```
Comment (Israel-Laguan) [inline pages/info/[memberCode].tsx:68]: Add a guard for this

File: pages/info/[memberCode].tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+          <>
+            <div className={styles['card-header']}>
+              {user.avatarUrl ? (
+                <Image
+                  src={user.avatarUrl}
+                  alt={`Avatar of ${user.memberInfo?.firstNames} ${user.memberInfo?.lastNames}`}
+                />
+              ) : (
+                <img
+                  src="https://img.icons8.com/external-tanah-basah-glyph-tanah-basah/48/null/external-user-user-tanah-basah-glyph-tanah-basah-7.png"
```
Comment (Israel-Laguan) [inline pages/info/[memberCode].tsx]: What about public/images/default-avatar.png as default here?

File: styles/MemberInfo.module.scss [hunk 8]
```diff
@@ -0,0 +1,54 @@
+@import 'colors', 'fonts';
+
+.container {
+  display: flex;
+  justify-content: center;
+
+  .card {
+    background: #fff;
```
Comment (Israel-Laguan) [inline styles/MemberInfo.module.scss]: use color in variables

File: styles/MemberInfo.module.scss [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+      padding: 1.5rem;
+
+      p {
+        font-size: 1rem;
+        margin: .5rem 0;
+      }
+    }
+
+    .card-body {
+      background: #f2f2f2;
```
Comment (Israel-Laguan) [inline styles/MemberInfo.module.scss]: use color in variables

**Review by Israel-Laguan — COMMENTED** [review]
Why there is a change on lock file? 
image

Please revert that change

File: pages/_app.tsx [hunk 10]
```diff
@@ -67,7 +67,7 @@ const DIDAdminPanel = ({ Component, pageProps: { ...pageProps } }: AppProps) =>
         <meta name="author" content="Israel Antonio Rosales Laguan" />
       </Head>
       <ApolloProvider client={client}>
-        {pathname === '/ingresar' ? (
+        {pathname === '/ingresar' || pathname === '/info/[memberCode]' ? (
```
Comment (heliomar-pena) [inline pages/_app.tsx:70]: @Eghinner please create a task to use this route in the APP to create the QR

Comment (Eghinner) [reply pages/_app.tsx:70]: Ehmm, direct change
https://github.com/Azordev/did-app/blob/82a2c29f5e874f5ed713ac46e1d94be1242cf555/src/pages/User/components/UserHeader/UserHeader.vue#L49

Comment (heliomar-pena) [reply pages/_app.tsx:70]: oh ok, perfect!

File: .stylelintrc.json (hunk 2 above)
Comment (e-azocar) [reply .stylelintrc.json:28]: This expect an number, array or null
image

File: pages/info/[memberCode].tsx [hunk 11]
```diff
@@ -68,9 +67,9 @@ export default MemberInfo
 export const getServerSideProps: GetServerSideProps = async context => {
   const { memberCode } = context.query
 
-  const { user, error } = await getMemberByMemberCode(memberCode as string)
+  const res = await getMemberByMemberCode(memberCode as string)
```
Comment (Israel-Laguan) [inline pages/info/[memberCode].tsx:70]: Is this "errors' being handled?

Comment (e-azocar) [reply pages/info/[memberCode].tsx:70]: Here I show error, and a return home link
image

**Comment by angelik0828** [comment]
## ✔️ When performing the validation, the change made was found, the tests carried out are detailed below

- By scanning the QR Code generated in the application by the user in his profile I can scan it and go to the user's data

imagen

**Review by angelik0828 — APPROVED** [review]

**Review by Israel-Laguan — APPROVED** [review]

**Pull request merged**